compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp

```diff
@@ -313,6 +313,38 @@ struct ElideUnusedOp : public OpRewritePattern<Op> {
   }
 };
 
+// Clones ops that prefer to be cloned directly.
+// This prevents us from splatting out a value and then cloning that (keeping
+// the memory live/etc) instead of just splatting it again on-demand.
+//
+// Example:
+//  %0 = stream.async.splat %c123_i32
+//  %1 = stream.async.clone %0
+// ->
+//  %1 = stream.async.splat %c123_i32
+template <typename Op>
+struct PropagateClonableOps : public OpRewritePattern<Op> {
```

Comment:
Hah! I wrote the exact same thing here https://github.com/openxla/iree/pull/13711/files#diff-bbfa2f4047a3479b6d1fc36b56be3cee1e0f5bb01f76a339825cda79acffc452R479 (well almost). No need to reuse for now... I can consolidate later.